src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -1088,6 +1088,25 @@ async function handleMessage(message: string) {
 }
 
 async function main(): Promise<void> {
+  // Since the functions run as attached processes they naturally inherit SIGINT
+  // sent to the functions emulator. We want them to ignore the first signal
+  // to allow for a clean shutdown.
+  let lastSignal = new Date().getTime();
```

Comment:
Optional: extract a helper function for this double signal logic (and reuse for `emulators:start`) as well

Reply:
Yeah it's not quite similar enough due to the different logging/messaging and also we try not to share too much code with the runtime, so non-DRY it is :-/